tutor/commands/compose.py

```diff
@@ -343,6 +341,39 @@ def bindmount_command(context: BaseComposeContext, service: str, path: str) -> N
     )
 
 
+@click.command(
+    name="copyfrom",
+    help="Copy files/folders from a container directory to the local filesystem.",
+)
+@click.argument("service")
+@click.argument("container_path")
+@click.argument(
+    "host_path",
+    type=click.Path(exists=True, dir_okay=True, file_okay=False, resolve_path=True),
+)
```

Comment:
In a posix shell, the semantics of `cp -r SRC DST` are:
1. if `$DST` exists, copy `$SRC` to `$DST/$(basename $SRC)`
2. else, copy `$SRC` directly to `$DST`

(aside, I had to look these rules up because they're so natural to me that I never actually thought about them :)

Your implementation does (1), but folks used to posix's `cp` might expect/want (2) to also work:
```
$ tutor dev copyfrom lms /openedx/venv edx-platform-venv
Usage: tutor dev copyfrom [OPTIONS] SERVICE CONTAINER_PATH HOST_PATH
Try 'tutor dev copyfrom --help' for help.

Error: Invalid value for 'HOST_PATH': Directory 'edx-platform-venv' does not exist.
```

If you'd rather leave it as-is, no worries. This works well enough:
```
$ tutor dev copyfrom lms /openedx/venv . && mv venv edx-platform-venv
```

Reply:
I pulled my hair a little but I found an acceptable solution to implement (2). Please see the revised implementation.

Reply:
Looks great.